### DISCUSSION TITLE
Renaming interfaces

### DISCUSSION THREAD

**Discussion by WagoL** [netbox-community/netbox#12360] [Q&A]
I would like to know how I can rename interfaces in bulk. I know there is regex option and [this ](https://constantpinger.home.blog/2022/05/13/renaming-interfaces-in-bulk-in-netbox/) works fine for virtual chassis where the original port is just prepended with a prefix.
But how do you tackle the problem for example with patch panels that has Port 1 to 24. That I want to rename to Port 25 to 48.

**Comment by candlerb** [comment]
I think you'll probably need to write a Custom Script for that, or a script which hits the REST API port by port. Or just do it by hand :-)

You might find some clues from the custom script [here](https://github.com/netbox-community/customizations/blob/master/scripts/multi_connect.py) which does something different: it creates multiple cables between pairs of devices.  However, the reason I mention it is that you can specify `Gig0/[1-24]` on one device and `Gig0/[25-48]` on another, it will expand the two sets independently, and make the connections between the corresponding ports.  You could write a script which does renaming instead.

It would be tricky if any of the destination ports overlapped, e.g. rename port 1-8 to port 5-12. In that case, you'd probably need to rename to temporary names, and then rename them a second time to final target names.
